packages/storefront-lwc/src/modules/commerce/headerBasket/headerBasket.js

```diff
@@ -4,15 +4,31 @@
     SPDX-License-Identifier: BSD-3-Clause
     For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 */
-import { LightningElement } from 'lwc';
+import { LightningElement, wire } from 'lwc';
 import { ShoppingBasket } from 'commerce/data';
+import { useQuery } from '@lwce/apollo-client';
+import gql from 'graphql-tag';
+
+const QUERY = gql`
+    query {
+        getBasketProductCount
+    }
+`;
 
 /**
  * Header basket component that should show up in the header
  */
 export default class HeaderBasket extends LightningElement {
     quantity = 0;
 
+    @wire(useQuery, {
+        query: QUERY,
+        lazy: false,
+    })
+    updateProductsTotal(response) {
+        this.quantity = response.data ? response.data.getBasketProductCount : 0;
+    }
+
     constructor() {
         super();
         ShoppingBasket.updateBasketListener(
```

Comment:
i dont think we need this

Reply:
headerBasket component needs to listen to the "update" (add, remove product) events on the basket. We need to bind these events to update the headerBasket product count.

Reply:
@abkap02 - Please update this when you are done changing the Singleton implementation for Basket.